iphone/cli/commands/_build.js

```diff
@@ -4229,15 +4241,24 @@ iOSBuilder.prototype.processTiSymbols = function processTiSymbols() {
 	if (this.target === 'simulator' || this.includeAllTiModules) {
 		var definesFile = path.join(this.platformPath, 'Classes', 'defines.h');
 
+		contents = fs.readFileSync(definesFile).toString();
```

Comment:
The file is now being read in twice. You only need to read in the file if `!this.runOnMainThread || this.useJSCore`. Please add a if block around this and conditionally check each specific flag like this:

```
var definesFile = path.join(this.platformPath, 'Classes', 'defines.h');

if (this.runOnMainThread && !this.useJSCore) {
	// just symlink the file
	if (!this.copyFileSync(definesFile, dest)) {
		this.logger.trace(__('No change, skipping %s', dest.cyan));
	}
	return;
}

contents = fs.readFileSync(definesFile).toString();
if (this.useJSCore) {
	contents += '\n#define USE_JSCORE_FRAMEWORK';
}
if (!this.runOnMainThread) {
	contents += '\n#define TI_USE_KROLL_THREAD';
}
```